sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -366,13 +366,6 @@ final class DataFrameWriter private[sql](df: DataFrame) {
       case (true, SaveMode.ErrorIfExists) =>
         throw new AnalysisException(s"Table $tableIdent already exists.")
 
-      case (true, SaveMode.Append) =>
```

Comment:
I consolidated the code paths.  Any writing to `HadoopFsRelations` now goes through `InsertIntoHadoopFsRelation`.